ir/shared/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -302,41 +285,30 @@ object Types {
      *  For `NullType` and `NothingType`, the char codes are `'N'` and `'E'`,
      *  respectively.
      */
-    val charCode: Char = tpe match {
-      case VoidType    => 'V'
-      case BooleanType => 'Z'
-      case CharType    => 'C'
-      case ByteType    => 'B'
-      case ShortType   => 'S'
-      case IntType     => 'I'
-      case LongType    => 'J'
-      case FloatType   => 'F'
-      case DoubleType  => 'D'
-      case NullType    => 'N'
-      case NothingType => 'E'
-    }
+    val charCode: Char = charCodeInit
+
+    // Make sure the `case class` does not produce a public copy method
+    private def copy(tpe: PrimTypeWithRef)(charCodeInit: Char, displayNameInit: String): PrimRef =
+      throw new Error("dead code")
```

Comment:
Actually, I wonder if it wouldn't be cleaner to just have a class with an unapply method instead?

Reply:
Probably. It does cause a slight overhead on unapplying, compared to a true case class, but it might not be relevant here.

Reply:
That breaks a few more things, but I guess it's fine. 🤷‍♂️